egs2/TEMPLATE/tts1/tts.sh

```diff
@@ -620,18 +620,32 @@ if [ ${stage} -le 7 ] && [ ${stop_stage} -ge 7 ]; then
     _feats_type="$(<${data_feats}/${train_set}/feats_type)"
 
     # NOTE(kamo): If feats_type=raw, vocoder_conf is unnecessary
-    if [ "${_feats_type}" == fbank ] || [ "${_feats_type}" == stft ]; then
+    _scp=wav.scp
+    _type=sound
+    if [ "${_feats_type}" = fbank ] || [ "${_feats_type}" = stft ]; then
         _opts+="--vocoder_conf n_fft=${n_fft} "
         _opts+="--vocoder_conf n_shift=${n_shift} "
         _opts+="--vocoder_conf win_length=${win_length} "
         _opts+="--vocoder_conf fs=${fs} "
+        _scp=feats.scp
+        _type=kaldi_ark
     fi
-    if [ "${_feats_type}" == fbank ]; then
+    if [ "${_feats_type}" = fbank ]; then
         _opts+="--vocoder_conf n_mels=${n_mels} "
         _opts+="--vocoder_conf fmin=${fmin} "
         _opts+="--vocoder_conf fmax=${fmax} "
     fi
 
+    # If we use teacher forcing or GST embedding in inference,
+    # we need to pass speech as well as text
+    # TODO(kan-bayashi): Make this part more smarter
+    _use_teacher_forcing="$(pyscripts/utils/get_yaml.py "${decode_config}" use_teacher_forcing)"
+    _use_gst="$(pyscripts/utils/get_yaml.py "${train_config}" tts_conf.use_gst)"
+    _use_speech=false
+    if [ "${_use_teacher_forcing,,}" = "true" ] || [ "${_use_gst,,}" = "true" ]; then
+        _use_speech=true
+    fi
```

Comment:
I think this part is not so good.
Do you have any idea? @kamo-naoyuki 
I want to add additional opts when the specific option is True in config.